### PR TITLE
API allows journalist can log in

### PR DESCRIPTION
**Pull request by pierre-1**
# [PT Story](https://www.pivotaltracker.com/story/show/171936153)
## User Story
```
As a journalist
In order to access the articles I have written
I would like to be able to log in
```

## Changes proposed in this pull request:
* Adds RSpec test for journalist can login
* Updates controller for article creation so only journalists can crate article
* Adds enum roles for user model to validate the presence of :role
* Updates RSpec session test to use roles for user and journalist
* Updates Factory Bot to create a user and journalist
* Updates seed file with user and journalist roles

## What we have learned working on this feature:
* Refreshed knowledge about user roles and conditioning actions on authentication
* How to create multiple user for a test to test user role restriction

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class Api::ArticlesController < ApplicationController
+  before_action :authenticate_user!, only: %i[create]
```
Comment (blake-futchi) [inline app/controllers/api/articles_controller.rb:2]: what does %i do?

File: db/migrate/20200403075915_add_role_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :role, :integer, default: 0
```
Comment (blake-futchi) [inline db/migrate/20200403075915_add_role_to_users.rb:3]: assuming default 0 refers to user role?

File: spec/requests/api/journalist_can_create_article_spec.rb [hunk 3]
```diff
@@ -1,5 +1,14 @@
 RSpec.describe 'POST /articles', type: :request do
-  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+  let(:journalist) { create(:user, role: 'journalist') }
+  let(:journalist_credentials) { journalist.create_new_auth_token }
+  let(:journalist_headers) do
+    { HTTP_ACCEPT: 'application/json' }.merge!(journalist_credentials)
```
Comment (blake-futchi) [inline spec/requests/api/journalist_can_create_article_spec.rb:5]: what does require merge do?

**Review by blake-futchi — COMMENTED** [review]
just a few questions for you guys!

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (pierre-1) [reply app/controllers/api/articles_controller.rb:2]: The journalist needs to be authenticated when he/she creates an article.

File: db/migrate/20200403075915_add_role_to_users.rb (hunk 2 above)
Comment (pierre-1) [reply db/migrate/20200403075915_add_role_to_users.rb:3]: Indeed!

File: spec/requests/api/journalist_can_create_article_spec.rb (hunk 3 above)
Comment (pierre-1) [reply spec/requests/api/journalist_can_create_article_spec.rb:5]: Similar logic about journalists having to be logged in to fill out the form.

**Pull request merged**